recipes-app/node-red-preinstalled-nodes/node-red-preinstalled-nodes_0.1.bb

```diff
@@ -51,6 +51,10 @@ python do_generate_package_json() {
 addtask generate_package_json before do_install
 
 do_install() {
+    # install package.json into rootfs
+    install -d ${D}/usr/lib/nodered/.node-red
+    install -m 0644 ${WORKDIR}/package.json ${D}/usr/lib/nodered/.node-red
+    # add symlink from persistent partition to file on rootfs
     install -d ${D}/${NODE_RED_HOME_DIR}/.node-red
-    install -m 0644 ${WORKDIR}/package.json ${D}/${NODE_RED_HOME_DIR}/.node-red
+    ln -sr ${D}/usr/lib/nodered/.node-red/package.json ${D}/${NODE_RED_HOME_DIR}/.node-red/
```

Comment:
Why do we need to install things into a separate dir under `/usr/lib`? Can't we use the homedir directly, as before?

Reply:
My understanding was that the `node-red-preinstalled-nodes` might be modified by the user at runtime. If that is not the case, we better directly install into the homedir.

Reply:
"Might" or rather "might **not** be modified"? Yes, they might be modified by users, specifically by triggering an update via the palette. That may even apply now to the GPIO node is it has been upstreamed.

Reply:
With the symlink in place, this file still cannot be modified. But if required by a user it can simply be replaced by a copy and from now on the user is responsible for maintaining state.

Reply:
Changed install logic. Now we install the `node-red-preinstalled-nodes` as dedicated package into the ro-rootfs on swupdate.

Reply:
With the new install pattern, `node-red-preinstalled-nodes` is now a simple debian meta-package, but not a NPM package anymore.